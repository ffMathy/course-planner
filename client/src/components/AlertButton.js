import React from "react";
import {
    Button,
    AlertDialog,
    AlertDialogContent,
    AlertDialogOverlay,
    AlertDialogBody,
    AlertDialogFooter,
    AlertDialogHeader,
    IconButton,
} from "@chakra-ui/core";

const AlertButton = ({ itemType, itemName, action, navigateTo, confirmFn }) => {
    const [isOpen, setIsOpen] = React.useState();
    const onClose = () => setIsOpen(false);
    const cancelRef = React.useRef();

    return (
        <>
            <IconButton icon="delete" variantColor="red" size="lg" onClick={() => setIsOpen(true)} />
            <AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
                <AlertDialogOverlay />
                <AlertDialogContent>
                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
                        {action} {itemType}
                    </AlertDialogHeader>

                    <AlertDialogBody>
                        Are you sure you want to {action && action.toLowerCase()} {itemName}? You can't undo this action afterwards.
                    </AlertDialogBody>

                    <AlertDialogFooter>
                        <Button ref={cancelRef} onClick={onClose}>
                            Cancel
                        </Button>
                        <Button
                            variantColor="red"
                            onClick={() => {
                                onClose();
                                confirmFn();
                            }}
                            ml={3}
                        >
                            Delete
                        </Button>
                    </AlertDialogFooter>
                </AlertDialogContent>
            </AlertDialog>
        </>
    );
};

export default AlertButton;
