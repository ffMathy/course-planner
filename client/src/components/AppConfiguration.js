import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { theme, ThemeProvider, CSSReset } from "@chakra-ui/core";

/**
 * Wrapper Component to wrap all the different configurations
 * the main App Component uses.
 */

const AppConfiguration = ({ children }) => {
    return (
        <ThemeProvider theme={customTheme}>
            <CSSReset />
            <Router>{children}</Router>
        </ThemeProvider>
    );
};

// See: https://chakra-ui.com/getting-started
const customTheme = {
    ...theme,
};

export default AppConfiguration;
