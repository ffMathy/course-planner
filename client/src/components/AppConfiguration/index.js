import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { theme, ThemeProvider, CSSReset } from "@chakra-ui/core";

/**
 * Wrapper Component to wrap all the different configurations
 * the main App Component uses.
 */

const AppConfiguration = ({ children }) => {
    return (
        <ThemeProvider theme={theme}>
            <CSSReset />
            <Router>{children}</Router>
        </ThemeProvider>
    );
};

export default AppConfiguration;
