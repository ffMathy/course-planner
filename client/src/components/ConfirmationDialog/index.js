import React from "react";
import {
    Button,
    AlertDialog,
    AlertDialogOverlay,
    AlertDialogContent,
    AlertDialogHeader,
    AlertDialogBody,
    AlertDialogFooter,
} from "@chakra-ui/core";

const ConfirmationDialog = ({ isOpen, onClose, confirm, itemType, itemName, action, navigateTo }) => {
    const cancelRef = React.useRef();

    return (
        <>
            <AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
                <AlertDialogOverlay />
                <AlertDialogContent>
                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
                        {action} {itemType}
                    </AlertDialogHeader>

                    <AlertDialogBody>
                        Are you sure you want to {action && action.toLowerCase()} {itemName}? You can't undo this action afterwards.
                    </AlertDialogBody>

                    <AlertDialogFooter>
                        <Button ref={cancelRef} onClick={onClose}>
                            Cancel
                        </Button>
                        <Button variantColor="red" onClick={confirm} ml={3}>
                            {action}
                        </Button>
                    </AlertDialogFooter>
                </AlertDialogContent>
            </AlertDialog>
        </>
    );
};

export default ConfirmationDialog;
