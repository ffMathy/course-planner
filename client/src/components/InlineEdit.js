import React from "react";
import { Editable, EditableInput, EditablePreview, Box, Text } from "@chakra-ui/core";

const InlineEdit = ({ onSubmit, onCancel, onChange, value, title, placeholder }) => {
    return (
        <Editable
            value={value}
            onSubmit={onSubmit}
            onCancel={onCancel}
            onChange={onChange}
            marginTop="8px"
            textAlign="center"
            fontSize="2xl"
            display="flex"
            flexDirection="row"
            justifyContent="center"
            alignContent="center"
            placeholder={placeholder}
        >
            <Box>
                <Text> {title}: </Text>
            </Box>
            <Box>
                <EditablePreview ml={3} height={10} />
            </Box>
            <Box>
                <EditableInput ml={3} height={10} />
            </Box>
        </Editable>
    );
};

export default InlineEdit;
