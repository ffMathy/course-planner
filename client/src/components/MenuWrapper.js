import React from "react";
import { Flex } from "@chakra-ui/core";
import { OptionsMenu, ConfirmationDialog } from ".";

const MenuWrapper = ({ item, itemType, setOpenConfirmationDialog, openConfirmationDialog, confirm, hasEdit, onEdit }) => {
    return (
        <Flex>
            <OptionsMenu confirm={confirm} hasEdit={hasEdit} onEdit={onEdit} onDelete={() => setOpenConfirmationDialog(true)} />
            <ConfirmationDialog
                item={item}
                itemType={itemType}
                action={"Delete"}
                isOpen={openConfirmationDialog}
                onClose={() => setOpenConfirmationDialog(false)}
                confirm={confirm}
            />
        </Flex>
    );
};

export default MenuWrapper;
