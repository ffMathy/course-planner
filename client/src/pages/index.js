import HomePage from "./HomePage";
import CoursesPage from "./CoursesPage";
import ProgrammesPage from "./Programmes/ProgrammesPage";
import Students from "./Students";

export { HomePage, CoursesPage, ProgrammesPage, Students };
