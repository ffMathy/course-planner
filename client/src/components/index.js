import AppConfiguration from "./AppConfiguration";
import OutlineButton from "./OutlineButton";
import ProgrammeCard from "./ProgrammeCard";
import SearchBar from "./SearchBar";
import ProgrammeRequirementsItem from "./ProgrammeRequirementsItem";
import ConfirmationDialog from "./ConfirmationDialog";

export { AppConfiguration, OutlineButton, ProgrammeCard, SearchBar, ProgrammeRequirementsItem, ConfirmationDialog };
