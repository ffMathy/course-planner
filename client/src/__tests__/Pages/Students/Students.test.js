import React from "react";
import renderer from "react-test-renderer";
import { AppConfiguration } from "../../../components";
import { Students } from "../../../pages";

test("Main Students Page", () => {
    const tree = renderer
        .create(
            <AppConfiguration>
                <Students />
            </AppConfiguration>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
