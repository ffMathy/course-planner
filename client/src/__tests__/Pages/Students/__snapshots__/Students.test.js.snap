// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Students Page 1`] = `
<div
  className="css-m5hsxh"
>
  <div
    className="css-mzeyh"
  >
    <div
      className="css-1u9fuhx"
    >
      <button
        aria-controls="menu-1"
        aria-expanded={false}
        aria-haspopup="menu"
        className="css-wk9st5"
        id="menubutton-2"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <svg
          aria-hidden={true}
          className="css-thzv76"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
      </button>
    </div>
    <div
      className="css-9jtr4b"
    >
      <p
        className="css-o10up6"
      >
        Students
      </p>
    </div>
    <div
      className="css-1evpibw"
    >
      <div
        className="css-9jn3o8"
      >
        <div
          className="css-yh0f2e"
        >
          <div
            className="css-1lads1q"
          >
            <div
              className="css-3lf2mu"
            >
              <div
                className="css-k6882c"
              >
                <svg
                  className="css-1xb1rov"
                  focusable={false}
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <input
                className="css-n5gi9c"
                onChange={[Function]}
                placeholder="Search Students"
                type="search"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-jeup5m"
      >
        <button
          className="css-1dad4i1"
          onClick={[Function]}
          type="button"
        >
          <p
            className="css-168qb02"
          >
            Add Student
          </p>
        </button>
      </div>
    </div>
    <div
      className="css-nc95w2"
    >
      <p>
         No Courses Found
      </p>
    </div>
  </div>
  <hr
    aria-orientation="vertical"
    className="css-lcuxz7"
  />
  <div
    className="css-mzeyh"
  >
    <div
      className="css-1qd02q1"
    >
      <div
        className="css-mzeyh"
      >
        <div
          className="css-1xhj18k"
        >
          <div
            className="css-1uiyeba"
          >
            <div
              className="css-k008qs"
            >
              <button
                aria-controls="menu-3"
                aria-expanded={false}
                aria-haspopup="menu"
                className="css-992tcj"
                id="menubutton-4"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                <svg
                  className="css-11orafl"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  type="button"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2S13.1 10 12 10zM18 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2S19.1 10 18 10zM6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2S7.1 10 6 10z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <p
            className="css-qvwayb"
          >
            Selected Student
          </p>
        </div>
        <div
          className="css-wbhiop"
        >
          <div
            className="css-gltekd"
          >
            <div
              className="css-0"
            >
              <p
                className="css-fzcsno"
              >
                 
                Name
                : 
              </p>
            </div>
            <div
              className="css-0"
            >
              <span
                className="css-1790c3i"
                onFocus={[Function]}
                tabIndex={0}
              >
                undefined
              </span>
            </div>
            <div
              className="css-0"
            />
          </div>
          <div
            className="css-gltekd"
          >
            <div
              className="css-0"
            >
              <p
                className="css-fzcsno"
              >
                 
                UPI
                : 
              </p>
            </div>
            <div
              className="css-0"
            >
              <span
                className="css-1790c3i"
                onFocus={[Function]}
                tabIndex={0}
              >
                undefined
              </span>
            </div>
            <div
              className="css-0"
            />
          </div>
          <div
            className="css-gltekd"
          >
            <div
              className="css-0"
            >
              <p
                className="css-fzcsno"
              >
                 
                ID
                : 
              </p>
            </div>
            <div
              className="css-0"
            >
              <span
                className="css-1790c3i"
                onFocus={[Function]}
                tabIndex={0}
              >
                undefined
              </span>
            </div>
            <div
              className="css-0"
            />
          </div>
          <div
            className="css-1h3rw7k"
          >
            <div
              className="css-0"
            >
              <p
                className="css-fzcsno"
              >
                 Year Level: 
              </p>
            </div>
            <div
              className="css-x6sard"
            >
              <select
                className="css-16ut77p"
                onChange={[Function]}
              >
                <option
                  value={1}
                >
                  Part 1
                </option>
                <option
                  value={2}
                >
                  Part 2
                </option>
                <option
                  value={3}
                >
                  Part 3
                </option>
                <option
                  value={4}
                >
                  Part 4
                </option>
                <option
                  value={5}
                >
                  Part 5
                </option>
              </select>
              <div
                className="css-1lepn2w"
              >
                <svg
                  aria-hidden="true"
                  className="css-12c4avn"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-cwx5qq"
        >
          <div
            className="css-1hyoz7m"
          >
            <p
              className="css-u152l3"
            >
              Plans
            </p>
            <a
              href="/plan/undefined/3"
              onClick={[Function]}
            >
              <button
                className="css-abregs"
                type="button"
              >
                <p
                  className="css-168qb02"
                >
                  Create Plan
                </p>
              </button>
            </a>
          </div>
          <div
            className="css-nb2vca"
          >
            <table
              className="css-10trp67"
              role="table"
            >
              <thead
                className="css-0"
              >
                <tr
                  className="css-0"
                  role="row"
                >
                  <th
                    className="css-1oifyfq"
                    colSpan={1}
                    role="columnheader"
                  >
                    <p
                      className="css-evrumw"
                    >
                      Name
                    </p>
                  </th>
                  <th
                    className="css-1oifyfq"
                    colSpan={1}
                    role="columnheader"
                  >
                    <p
                      className="css-evrumw"
                    >
                      Programme
                    </p>
                  </th>
                  <th
                    className="css-1oifyfq"
                    colSpan={1}
                    role="columnheader"
                  >
                    <p
                      className="css-evrumw"
                    >
                      Start Year
                    </p>
                  </th>
                  <th
                    className="css-1oifyfq"
                    colSpan={1}
                    role="columnheader"
                  >
                    <p
                      className="css-evrumw"
                    >
                      End Year
                    </p>
                  </th>
                </tr>
              </thead>
              <tbody
                className="css-0"
                role="rowgroup"
              >
                <tr
                  className="css-0"
                  role="row"
                >
                  <td
                    className="css-1i8kqc6"
                    role="cell"
                  >
                    <p
                      className="css-6ewox4"
                    >
                      Plan1
                    </p>
                  </td>
                  <td
                    className="css-1i8kqc6"
                    role="cell"
                  >
                    <p
                      className="css-6ewox4"
                    >
                      BE (Hons) SE
                    </p>
                  </td>
                  <td
                    className="css-1i8kqc6"
                    role="cell"
                  >
                    <p
                      className="css-6ewox4"
                    >
                      2018
                    </p>
                  </td>
                  <td
                    className="css-1i8kqc6"
                    role="cell"
                  >
                    <p
                      className="css-6ewox4"
                    >
                      2020
                    </p>
                  </td>
                </tr>
                <tr
                  className="css-0"
                  role="row"
                >
                  <td
                    className="css-1i8kqc6"
                    role="cell"
                  >
                    <p
                      className="css-6ewox4"
                    >
                      Plan2
                    </p>
                  </td>
                  <td
                    className="css-1i8kqc6"
                    role="cell"
                  >
                    <p
                      className="css-6ewox4"
                    >
                      BE (Hone) SE
                    </p>
                  </td>
                  <td
                    className="css-1i8kqc6"
                    role="cell"
                  >
                    <p
                      className="css-6ewox4"
                    >
                      2018
                    </p>
                  </td>
                  <td
                    className="css-1i8kqc6"
                    role="cell"
                  >
                    <p
                      className="css-6ewox4"
                    >
                      2020
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
